backend/benefit/companies/api/v1/views.py

```diff
@@ -0,0 +1,62 @@
+from companies.api.v1.serializers import CompanySerializer
+from companies.models import Company
+from companies.services import get_or_create_company_with_business_id
+from companies.tests.data.company_data import DUMMY_COMPANY_DATA
+from django.conf import settings
+from django.db import transaction
+from requests.exceptions import HTTPError
+from rest_framework import status
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+
+class GetCompanyView(APIView):
+    """
+    API View to retrieve company info using the YTJ API integration.
+    """
+
+    @property
+    def ytj_api_error(self):
+        return Response(
+            "YTJ API is under heavy load or no company found with the given business id",
+            status.HTTP_404_NOT_FOUND,
+        )
+
+    @transaction.atomic
+    def get_mock(self, request: Request, format: str = None) -> Response:
+        # This mocked get method will be used for testing purposes for the frontend.
+        session_id = request.META.get("HTTP_SESSION_ID")
+        if session_id == "-1":
+            return self.ytj_api_error
+
+        company = Company(**DUMMY_COMPANY_DATA)
+        company_data = CompanySerializer(company).data
+
+        return Response(company_data)
+
+    @transaction.atomic
+    def get(self, request: Request, format: str = None) -> Response:
+        if settings.MOCK_FLAG:
+            return self.get_mock(request, format)
+        business_id = request.META.get("HTTP_BUSINESS_ID") or "0877830-0"
```

Comment:
Is 0877830-0 there by accident? (I. Haanpää Oy)

Reply:
This is something I used for testing purposes for YTJ.

Reply:
Yeah I think we can remove it in our Benefit, I don't think we'll need this in our client